sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/UtilsImpl.java

```
@@ -0,0 +1,250 @@
+        credentialPolicies.add(loggingPolicy);
+
+        ContainerRegistryTokenService tokenService = new ContainerRegistryTokenService(
+            credential,
+            audience,
+            endpoint,
+            serviceVersion,
+            new HttpPipelineBuilder()
+                .policies(credentialPolicies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build(),
+            JacksonAdapter.createDefaultSerializerAdapter());
+
+        ContainerRegistryCredentialsPolicy credentialsPolicy = new ContainerRegistryCredentialsPolicy(tokenService);
+        policies.add(credentialsPolicy);
+
+        policies.add(loggingPolicy);
+        HttpPipeline httpPipeline =
+            new HttpPipelineBuilder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+        return httpPipeline;
+    }
+
+    private static ArrayList<HttpPipelinePolicy> clone(ArrayList<HttpPipelinePolicy> policies) {
+        ArrayList<HttpPipelinePolicy> clonedPolicy = new ArrayList<>();
+        for (HttpPipelinePolicy policy:policies) {
+            clonedPolicy.add(policy);
+        }
+
+        return clonedPolicy;
+    }
+
+    /**
+     * This method computes the digest for the buffer content.
+     * Docker digest is a SHA256 hash of the docker image content and is deterministic based on the image build.
+     * @param buffer The buffer containing the image bytes.
+     * @return SHA-256 digest for the given buffer.
+     */
+    public static String computeDigest(ByteBuffer buffer) {
+        int startPosition = buffer.position();
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            md.update(buffer);
+            byte[] digest = md.digest();
+            return "sha256:" + byteArrayToHex(digest);
+
+        } catch (NoSuchAlgorithmException e) {
```

Should this exception really be swallowed?

---

No, I have created an issue to track fixing this part. [https://github.com/Azure/azure-sdk-for-java/issues/27088](https://github.com/Azure/azure-sdk-for-java/issues/27088)